### PR TITLE
update bmclib/bmc-common/ironlib/ctrl/rivets

### PR DESCRIPTION
**Pull request by DoctorVin**
#### What does this PR do
move everything to be consistent on metal-toolbox/bmclib and metal-toolbox/bmc-common. some less-trivial changes in terms of getting an inventory from FleetDB via the GetServerInventory API instead of getting each bit piecemeal and stitching it all together.

### CODE REVIEW

**Review by coffeefreak101 — APPROVED** [review]

File: internal/store/fleetdb.go [hunk 1]
```diff
@@ -163,68 +164,33 @@ func (s *FleetDBAPI) AssetByID(ctx context.Context, id string) (*rtypes.Server,
...
-		s.registerMetric("GetCredential")
+		registerMetric("GetCredential")
 
 		return nil, errors.Wrap(ErrServerserviceQuery, "GetCredential: "+err.Error())
 	}
 
 	asset.BMCUser = credential.Username
 	asset.BMCPassword = credential.Password
 
-	// query the server object
```
Comment (jakeschuurmans) [inline internal/store/fleetdb.go:179]: What exactly is this about?

**Review by jakeschuurmans — APPROVED** [review]
just curious about the deleted file and deleted sections from fleetdb.go

File: internal/store/fleetdb.go (hunk 1 above)
Comment (DoctorVin) [reply internal/store/fleetdb.go:179]: What you see here is conversion of FleetDB data to a `rivets.Server`: 

1. Get a server object in FleetDB "form"
2. Get its bmc address via the attributes retrieved from the server record
3. Get the server vendor/model/serial from the attributes
4. Get the components
5. Stitch it all into the `rivets.Server` structure.

The new way is:

1.  "call GetServerInventory"
2. profit

This depends on a deploy of FleetDB (cf. https://github.com/metal-toolbox/fleetdb/pull/122) but it works fine in the sandbox using FleetDB `main`.

Comment (DoctorVin) [reply internal/store/fleetdb.go:179]: We don't need code to decrypt FleetDB Attributes into components if we call an API that does that for you, so I could delete the attributes code.

**Pull request merged**